src/flood_model/bg_flood_model.py

```diff
@@ -398,16 +398,13 @@ def run_bg_flood_model(
     # Get the valid BG-Flood Model directory
     bg_flood_dir = get_valid_bg_flood_dir()
     # Get the file path of the Hydro DEM for the catchment area
-    hydro_dem_path = dem_metadata_in_db.get_catchment_hydro_dem_filepath(engine, catchment_area)
-    # Use the resolution of the Hydro DEM if it is not provided
-    if resolution is None:
-        _, resolution = dem_metadata_in_db.get_hydro_dem_data_and_resolution(engine, catchment_area)
+    hydro_dem_path, _, _, resolution = get_dem_by_geometry(engine, catchment_area)
```

Comment:
```suggestion
    hydro_dem_path, _, _, resolution_of_dem = get_dem_by_geometry(engine, catchment_area)
    if resolution is None:
        resolution = resolution_of_dem
```
Idea of what @caseyli92 may be referring to, but disagreements on where the resolution should come from.
Resolution from either DEM or from user input for specifying output resolution or from database table

Reply:
done